js/views/standalone_registration_view.js

```diff
@@ -123,6 +126,38 @@
       const language = this.$('#mnemonic-display-language').val();
       this.showProfilePage(mnemonic, language);
     },
+    async registerSecondaryDevice() {
+      const mnemonic = this.$('#mnemonic-display').text();
+      const language = this.$('#mnemonic-display-language').val();
+      const primaryPubKey = this.$('#primary-pubkey').val();
+      this.$('.standalone-secondary-device #error').hide();
+      Whisper.events.on('secondaryDeviceRegistration', () => {
```

Comment:
Consecutive calls to `registerSecondaryDevice` will register this event multiple times. Replacing this with `.once` will ensure that we don't have too many long lasting event listeners but it still doesn't fix the multiple event registrations for the function.

A solution would be to move the callback to a named function and then call `Whisper.events.remove('secondaryDeviceRegistration', handlerFunction)` before calling the event registration

Reply:
Updated, does that fix it?